In [1]:
# Dependencies
import pandas as pd
import plotly.express as plt
import os
import statsmodels.api as sm
import numpy as np

In [2]:
#import data
bikecounts_file = os.path.join('Resources', 'bikecounts.csv')
bikecounts_df = pd.read_csv(bikecounts_file)
bikecounts_df['Day'] = pd.to_datetime(bikecounts_df['Day'])
bikecounts_df#.head()

Unnamed: 0        Date        Day  High Temp (°F)  Low Temp (°F)  \
0            0  2016-04-01 2016-04-01            78.1           66.0   
1            1  2016-04-02 2016-04-02            55.0           48.9   
2            2  2016-04-03 2016-04-03            39.9           34.0   
3            3  2016-04-04 2016-04-04            44.1           33.1   
4            4  2016-04-05 2016-04-05            42.1           26.1   
5            5  2016-04-06 2016-04-06            45.0           30.0   
6            6  2016-04-07 2016-04-07            57.0           53.1   
7            7  2016-04-08 2016-04-08            46.9           44.1   
8            8  2016-04-09 2016-04-09            43.0           37.9   
9            9  2016-04-10 2016-04-10            48.9           30.9   
10          10  2016-04-11 2016-04-11            62.1           46.0   
11          11  2016-04-12 2016-04-12            57.0           45.0   
12          12  2016-04-13 2016-04-13            57.0           39.9   
13          13  2016-04-14 2016-04-14            62.1           44.6   
14          14  2016-04-15 2016-04-15            64.0           44.1   
15          15  2016-04-16 2016-04-16            66.0           45.0   
16          16  2016-04-17 2016-04-17            73.9           46.0   
17          17  2016-04-18 2016-04-18            81.0           52.0   
18          18  2016-04-19 2016-04-19            71.1           63.0   
19          19  2016-04-20 2016-04-20            68.0           50.0   
20          20  2016-04-21 2016-04-21            71.1           50.0   
21          21  2016-04-22 2016-04-22            78.1           63.0   
22          22  2016-04-23 2016-04-23            70.0           61.0   
23          23  2016-04-24 2016-04-24            68.0           48.0   
24          24  2016-04-25 2016-04-25            66.9           54.0   
25          25  2016-04-26 2016-04-26            60.1           46.9   
26          26  2016-04-27 2016-04-27            62.1           46.9   
27          27  2016-04-28 2016-04-28            57.9           48.0   
28          28  2016-04-29 2016-04-29            57.0           46.9   
29          29  2016-04-30 2016-04-30            64.0           48.0   

   Precipitation  Brooklyn Bridge  Manhattan Bridge  Williamsburg Bridge  \
0           0.01           1704.0              3126               4115.0   
1           0.15            827.0              1646               2565.0   
2           0.09            526.0              1232               1695.0   
3       0.47 (S)            521.0              1067               1440.0   
4              0           1416.0              2617               3081.0   
5              0           1885.0              3329               3856.0   
6           0.09           1276.0              2581               3282.0   
7           0.01           1982.0              3455               4113.0   
8           0.09            504.0               997               1507.0   
9              0           1447.0              2387               3132.0   
10          0.01           2005.0              3791               4334.0   
11           0.2           1045.0              2178               2762.0   
12             0           2840.0              5395               5995.0   
13             0           2861.0              5309               6030.0   
14             0           2770.0              5072               5816.0   
15             0           2384.0              4316               5624.0   
16             0           3147.0              4969               5867.0   
17             0           3871.0              6823               7432.0   
18             0           3501.0              6951               7834.0   
19             0           3450.0              6574               7639.0   
20             0           3436.0              6452               7426.0   
21             T           2975.0              4907               6093.0   
22          0.16       

In [3]:
# Removing symbols/characters and changing  precipitation to float
def clear_characters(number):
    for chara in ['T', '(S)', ' ']:
        number = number.replace(chara, '')
    return number.strip()

In [4]:
bikecounts_df['Precipitation'] = bikecounts_df['Precipitation'].apply(clear_characters)
bikecounts_df['Precipitation'] = bikecounts_df['Precipitation'].replace('',0).astype('float64')
bikecounts_df#.head()

Unnamed: 0        Date        Day  High Temp (°F)  Low Temp (°F)  \
0            0  2016-04-01 2016-04-01            78.1           66.0   
1            1  2016-04-02 2016-04-02            55.0           48.9   
2            2  2016-04-03 2016-04-03            39.9           34.0   
3            3  2016-04-04 2016-04-04            44.1           33.1   
4            4  2016-04-05 2016-04-05            42.1           26.1   
5            5  2016-04-06 2016-04-06            45.0           30.0   
6            6  2016-04-07 2016-04-07            57.0           53.1   
7            7  2016-04-08 2016-04-08            46.9           44.1   
8            8  2016-04-09 2016-04-09            43.0           37.9   
9            9  2016-04-10 2016-04-10            48.9           30.9   
10          10  2016-04-11 2016-04-11            62.1           46.0   
11          11  2016-04-12 2016-04-12            57.0           45.0   
12          12  2016-04-13 2016-04-13            57.0           39.9   
13          13  2016-04-14 2016-04-14            62.1           44.6   
14          14  2016-04-15 2016-04-15            64.0           44.1   
15          15  2016-04-16 2016-04-16            66.0           45.0   
16          16  2016-04-17 2016-04-17            73.9           46.0   
17          17  2016-04-18 2016-04-18            81.0           52.0   
18          18  2016-04-19 2016-04-19            71.1           63.0   
19          19  2016-04-20 2016-04-20            68.0           50.0   
20          20  2016-04-21 2016-04-21            71.1           50.0   
21          21  2016-04-22 2016-04-22            78.1           63.0   
22          22  2016-04-23 2016-04-23            70.0           61.0   
23          23  2016-04-24 2016-04-24            68.0           48.0   
24          24  2016-04-25 2016-04-25            66.9           54.0   
25          25  2016-04-26 2016-04-26            60.1           46.9   
26          26  2016-04-27 2016-04-27            62.1           46.9   
27          27  2016-04-28 2016-04-28            57.9           48.0   
28          28  2016-04-29 2016-04-29            57.0           46.9   
29          29  2016-04-30 2016-04-30            64.0           48.0   

    Precipitation  Brooklyn Bridge  Manhattan Bridge  Williamsburg Bridge  \
0            0.01           1704.0              3126               4115.0   
1            0.15            827.0              1646               2565.0   
2            0.09            526.0              1232               1695.0   
3            0.47            521.0              1067               1440.0   
4            0.00           1416.0              2617               3081.0   
5            0.00           1885.0              3329               3856.0   
6            0.09           1276.0              2581               3282.0   
7            0.01           1982.0              3455               4113.0   
8            0.09            504.0               997               1507.0   
9            0.00           1447.0              2387               3132.0   
10           0.01           2005.0              3791               4334.0   
11           0.20           1045.0              2178               2762.0   
12           0.00           2840.0              5395               5995.0   
13           0.00           2861.0              5309               6030.0   
14           0.00           2770.0              5072               5816.0   
15           0.00           2384.0              4316               5624.0   
16           0.00           3147.0              4969               5867.0   
17           0.00           3871.0              6823               7432.0   
18           0.00           3501.0              6951               7834.0   
19           0.00           3450.0              6574               7639.0   
20           0.00           3436.0              6452               7426.0   
21           0.00           2975.0              4907               6093.0   


HYPOTHESIS:-
The number of bikes crossing brooklyn bridge is directly proportional to precipitation.

NULL HYPOTHESIS:-
There is no relation between the precipitation and the number of bikes crossing Brooklyn Bridge.

Test:-
The test is being performed with an alpha value of 0.05. Hence, a pvalue less than 0.05 will result in the null hypothesis being rejected.

In [5]:
# Select independent terms, and dependent terms.
dependent_variable = bikecounts_df['Brooklyn Bridge']
independent_variable = bikecounts_df['Precipitation']

X = sm.add_constant(independent_variable)
model = sm.OLS(dependent_variable, X)
results = model.fit()

In [6]:
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:        Brooklyn Bridge   R-squared:                       0.359
Model:                            OLS   Adj. R-squared:                  0.337
Method:                 Least Squares   F-statistic:                     15.71
Date:                Fri, 22 Oct 2021   Prob (F-statistic):           0.000463
Time:                        20:18:54   Log-Likelihood:                -242.48
No. Observations:                  30   AIC:                             489.0
Df Residuals:                      28   BIC:                             491.8
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
const          2571.0184    166.446     15.447

Result:-

1)The P-value is found to be from the test 0.000463 which is less than the alpha value. Therefore the null hypothesis can be rejected.
2)The R-squared value is positive (0.359) so it can be established that there is a direct relationship between the number of bikes crossing the brooklyn bridge and the precipitation.